packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/handlers/GetOmniboxCommandsEventHandler.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import reactor.core.publisher.Sinks;
+
+/**
+ * The event handler to retrieve omnibox commands in the context of an editing context.
+ *
+ * @author gcoutable
+ */
+@Service
+public class GetOmniboxCommandsEventHandler implements IEditingContextEventHandler {
+
+    private final IObjectService objectService;
+
+    private final ICollaborativeMessageService messageService;
+
+    private final IQueryService queryService;
+
+    private final Counter counter;
+
+    public GetOmniboxCommandsEventHandler(IObjectService objectService, ICollaborativeMessageService messageService, IQueryService queryService, MeterRegistry meterRegistry) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.messageService = Objects.requireNonNull(messageService);
+        this.queryService = Objects.requireNonNull(queryService);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        return input instanceof GetOmniboxCommandsInput;
+    }
+
+    @Override
+    public void handle(Sinks.One<IPayload> payloadSink, Sinks.Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        List<Message> messages = List.of(new Message(this.messageService.invalidInput(input.getClass().getSimpleName(), CreateChildInput.class.getSimpleName()),
+                MessageLevel.ERROR));
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), input);
+        IPayload payload = null;
+
+        if (input instanceof GetOmniboxCommandsInput getOmniboxCommandsInput) {
+            List<OmniboxCommand> omniboxCommands = new ArrayList<>();
+
+            var queryBasedObjectsInput = new QueryBasedObjectsInput(UUID.randomUUID(), "aql:editingContext.allContents()->select(e | e.name.contains('" + getOmniboxCommandsInput.query() + "'))", Map.of("editingContext", editingContext));
+            var objectsPayload = this.queryService.execute(editingContext, queryBasedObjectsInput);
+
+            if (objectsPayload instanceof QueryBasedObjectsSuccessPayload queryBasedObjectsSuccessPayload) {
+                queryBasedObjectsSuccessPayload.result().forEach(object -> {
+                    var id = this.objectService.getId(object);
+                    var label = this.objectService.getLabel(object);
+                    var kind = this.objectService.getKind(object);
+                    var iconURL = this.objectService.getImagePath(object);
+                    omniboxCommands.add(new OmniboxCommand(UUID.fromString(id), label, kind, iconURL, ""));
```

Comment:
- The id of the object will not be an UUID in various use cases
- Add a description indicating that the element will be selected something like "Click to select the object"

Reply:
Done